types/src/protocol_version.rs

```diff
@@ -149,6 +163,20 @@ impl fmt::Display for ProtocolVersion {
     }
 }
 
+#[cfg(feature = "std")]
+impl JsonSchema for ProtocolVersion {
```

Comment:
I think this is incorrect.  This is specifying the JSON format as a string, but because we just derive `Serialize` here and for `SemVer` it won't be just a string, it'll be a JSON object.

I think we should follow the example of e.g. `Timestamp` where we have [custom impls for `Serialize` and `Deserialize`](https://github.com/CasperLabs/casper-node/blob/master/node/src/types/timestamp.rs#L158-L178) and then we can just derive the `JsonSchema` with [an extra attribute](https://github.com/CasperLabs/casper-node/blob/master/node/src/types/timestamp.rs#L37) to tell it we use string for JSON.  That should be applied to the `SemVer` type, and here we should probably use [`#[serde(flatten)]` ](https://serde.rs/attr-flatten.html) like this:
```
pub struct ProtocolVersion(#[serde(flatten)] SemVer);
```

I _think_ that will give the best results for JSON-encoding these, but we should check how this looks before merging this PR, and we should also eyeball the generated JSON-schema.

Reply:
A, ok. Thanks for the pointers.

Reply:
Did it a bit differently but here's the result:
```json
"header": {
    "accumulated_seed": "4cda5b116a1b66f12b77d106c2120a211c4542fdcdcb49d8560eb7fc43550398",
    "body_hash": "6a34d9ae04ed621e0ef8e460cb24b4dae9ec116ce82cb542ab5a51b6fb9953d3",
    "era_end": null,
    "era_id": 1,
    "height": 11,
    "parent_hash": "a0206acac97fb834ce9ef2f7205cb8e5edad294b83250e920846e712a98f4425",
    "protocol_version": "1.0.0",
    "random_bit": false,
    "state_root_hash": "eafd38f29dce7e17bafee6c08d1e165d8859fc15ccbe07a9f2d762f3a1966ebb",
    "timestamp": "2021-02-16T14:04:00.896Z"
  }
```

Reply:
and schema:
```json
"protocol_version": {
            "$ref": "#/components/schemas/ProtocolVersion"
          }
```

Reply:
I'm assuming this is done and resolving the comment.